File: src/main/java/it/tylframework/data/mongo/common/Footprint.java
package it.tylframework.data.mongo.common;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.joda.time.DateTime;
import org.springframework.data.annotation.*;
import org.springframework.data.domain.Auditable;

/**
 * Created with IntelliJ IDEA.
 * User: marco
 * Date: 18/11/14
 * Time: 22:37
 */
@Data
@EqualsAndHashCode(callSuper=false)
public class Footprint implements Auditable<Signature,String> {

    @CreatedBy
    private Signature createdBy;

    @CreatedDate
    private DateTime createdDate;

    @LastModifiedBy
    private Signature lastModifiedBy;

    @LastModifiedDate
    private DateTime lastModifiedDate;

    @Version
    private Long version;

    @Id
    private String id;

    @Override
    public boolean isNew() {
        return (id == null);
    }

}

File: src/main/java/it/tylframework/data/mongo/common/Footprint.kt
package it.tylframework.data.mongo.common

import java.util.Date
import org.springframework.data.annotation.Id

/**
 * Created by marco on 24/08/14.
 */
open class Footprint(
        val created_by: Signature,
        val created_on: Date,
        var updated_by: Signature,
        var updated_on: Date) {

    [Id]
    val id:String?=null
}